### PR TITLE
View2

### PR DESCRIPTION
**Pull request by michelececcacci**

### CODE REVIEW

**Comment by michelececcacci** [comment]
@smagnaniuni ready for merge

File: src/main/java/homer/api/DeviceView.java [hunk 1]
```diff
@@ -5,18 +5,18 @@
  * 
  * @param <S> The device state.
```
Comment (smagnaniuni) [inline src/main/java/homer/api/DeviceView.java:6]: to remove

File: src/main/java/homer/api/DeviceView.java [hunk 2]
```diff
@@ -5,18 +5,18 @@
...
  */
-public interface DeviceView<S> {
+public interface DeviceView {
 
     /**
      * 
      * @return Device information to be displayed visually.
      */
-    S getState();
+    DeviceState getState();
```
Comment (smagnaniuni) [inline src/main/java/homer/api/DeviceView.java:14]: Should we really have a `getState()` in the view? If I understood mvc correctly, this view should only have the `setState()` that is called by the controller

Comment (michelececcacci) [reply src/main/java/homer/api/DeviceView.java:14]: >Should we really have a getState() in the view? If I understood mvc correctly, this view should only have the setState() that is called by the controller

how do we deal with incoming updates then?

File: src/main/java/homer/api/DeviceView.java (hunk 1 above)
Comment (michelececcacci) [reply src/main/java/homer/api/DeviceView.java:6]: right

File: src/main/java/homer/api/DeviceView.java (hunk 2 above)
Comment (michelececcacci) [reply src/main/java/homer/api/DeviceView.java:14]: actually, i think you're right

**Pull request merged**